string regex
---
string topicNameTopicAdded
string topicNameTopicRemoved
string[] existingTopics